scripts/snfoundryup

```diff
@@ -0,0 +1,559 @@
+  # See https://github.com/rust-lang/rustup.rs/issues/587.
+  if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
+    if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+      # At least one processor does not have NEON.
+      _cputype=arm
+    fi
+  fi
+
+  _arch="${_cputype}-${_ostype}"
+
+  RETVAL="$_arch"
+}
+
+resolve_version() {
+  local _requested_version=$1
+  local _requested_ref=$2
+
+  local _response
+
+  say "retrieving $_requested_version version from ${REPO}..."
+  _response=$(ensure curl -# -Ls -H 'Accept: application/json' "${REPO}/releases/${_requested_ref}")
+  if [ "{\"error\":\"Not Found\"}" = "$_response" ]; then
+    err "version $_requested_version not found"
+  fi
+
+  RETVAL=$(echo "$_response" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+}
+
+create_install_dir() {
+  local _requested_version=$1
+
+  local _installdir="${INSTALL_ROOT}/${_requested_version}"
+
+  if [ -d "$_installdir" ]; then
+    ensure rm -rf "$_installdir"
+    say "removed existing snforge and sncast installation at ${_installdir}"
+  fi
+
+  ensure mkdir -p "$_installdir"
+
+  RETVAL=$_installdir
+}
+
+download() {
+  local _resolved_version=$1
+  local _arch=$2
+  local _installdir=$3
+  local _tempdir=$4
+
+  local _tarball="starknet-foundry-${_resolved_version}-${_arch}.tar.gz"
+  local _url="${REPO}/releases/download/${_resolved_version}/${_tarball}"
+  local _dl="$_tempdir/starknet-foundry.tar.gz"
+
+  say "downloading ${_tarball}..."
+
+  ensure curl -# -fLS -o "$_dl" "$_url"
+  ensure tar -xz -C "$_installdir" --strip-components=1 -f "$_dl"
+}
+
+create_symlinks() {
+  local _installdir=$1
+  for binary in $BINARIES; do
+    local _binary="${_installdir}/bin/${binary}"
+    local _symlink="${LOCAL_BIN}/${binary}"
+
+    ensure mkdir -p "$LOCAL_BIN"
+    ensure ln -fs "$_binary" "$_symlink"
+    ensure chmod u+x "$_symlink"
+    say "created symlink ${_symlink} -> ${_binary}"
+  done
+}
+
+add_local_bin_to_path() {
+  local _profile
+  local _pref_shell
+  case ${SHELL:-""} in
+  */zsh)
+    _profile=$HOME/.zshrc
+    _pref_shell=zsh
+    ;;
+  */ash)
+    _profile=$HOME/.profile
+    _pref_shell=ash
+    ;;
+  */bash)
+    _profile=$HOME/.bashrc
+    _pref_shell=bash
+    ;;
+  */fish)
+    _profile=$HOME/.config/fish/config.fish
+    _pref_shell=fish
+    ;;
+  *)
+    err "could not detect shell, manually add '${LOCAL_BIN_ESCAPED}' to your PATH."
+    ;;
+  esac
+
+  echo >>"$_profile" && echo "export PATH=\"\$PATH:${LOCAL_BIN_ESCAPED}\"" >>"$_profile"
+  echo \
+    "Detected your preferred shell is ${_pref_shell} and added '${LOCAL_BIN_ESCAPED}' to PATH." \
+    "Run 'source ${_profile}' or start a new terminal session to use Starknet Foundry."
+}
+
+clone_and_build() {
+  local _commit_hash=$1
+  local _tempdir
+
+  if ! _tempdir="$(ensure mktemp -d)"; then
+    exit 1
+  fi
+
+  say "Cloning repository at commit $_commit_hash..."
+  ensure git clone "$REPO" $_tempdir
+  ensure cd "$_tempdir"
```

Comment:
pushd and popd are sadly not part of sh shell

Reply:
Ahh, you're right, I forgot about it. Sorry for the confusion then.

Still maybe it would be a good idea to return to the previous directory in this function (eg `cd -` ?) I know that this function is used only at the end of `c` case for building from commit hash so it does not impact anything the moment, but I'm worried it might in the future when someone wants to use this function for something else (and the directories are mixed up). wdyt?

Reply:
I think your description is the right way to handle it, there maybe unwanted side effects. Adding a `cd -` to the end now

Reply:
updated [here](https://github.com/foundry-rs/starknet-foundry/pull/1041/commits/355011f6986699701b300e380a779c8e6f424389)